File: src/test/java/numberofones/SolutionTest.java
package numberofones;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

import static org.assertj.core.api.Assertions.assertThat;

class SolutionTest {

    private final Solution solution = new Solution();

    @ParameterizedTest
    @CsvSource(value = {"11:3", "128:1", "2147483645:30"}, delimiter = ':')
    public void testOne(String input, String output) {
        int count = solution.hammingWeight(Integer.parseInt(input));
        assertThat(count).isEqualTo(Integer.valueOf(output).intValue());
    }
}

File: src/test/kotlin/dev/sanket/validanagram/SolutionTest.kt
package dev.sanket.validanagram

import assertk.assertThat
import assertk.assertions.isEqualTo
import dev.sanket.dev.sanket.validanagram.Solution
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource

class SolutionTest {

    private val solution = Solution()

    @ParameterizedTest
    @CsvSource(value = ["ab:abc:false", "anagram:nagaram:true"], delimiter = ':')
    fun shouldPass(first: String, second: String, isValid: Boolean) {
        assertThat(solution.isAnagram(first, second)).isEqualTo(isValid)
    }
}